name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/RaiffeisenBankgruppePDFExtractor.java

```diff
@@ -1249,4 +1252,20 @@ protected BigDecimal asExchangeRate(String value)
 
         return ExtractorUtils.convertToNumberBigDecimal(value, Values.Share, language, country);
     }
+
+    private void setDateAndAmountForAccountStatement(AccountTransaction t, ParsedData v)
+    {
+        if (v.get("nr").compareTo("01") == 0 && Integer.parseInt(v.get("month")) < 3)
+        {
+            int year = Integer.parseInt(v.get("year")) + 1;
+            t.setDateTime(asDate(v.get("day") + "." + v.get("month") + "." + year));
+        }
+        else
+        {
+            t.setDateTime(asDate(v.get("day") + "." + v.get("month") + "." + v.get("year")));
+        }
+
+        t.setCurrencyCode(v.get("currency"));
+        t.setAmount(asAmount(v.get("amount")));
+    }
```

Comment:
The source code is also inconsistent here.
Either we apply the determination of the year to all transaction extractions or to none.

Furthermore, we should only create an "auxiliary class" that correctly reflects the name of the class and what it does. This means that if we use the auxiliary class, we should not use "date" and "amount" in the name and apply them. These are different treatments.

Remove the "currency and amount extraction" and leave them as they were and use/rename the function
"setDateAndAmountForAccountStatement" to "DateTransactionHelper"
If these are then applied to all transactions.

Translated with DeepL.com (free version)

Reply:
OK, but should I use the already existing class "ExtractorUtils" instead of creating a new one "DateTransactionHelper"?

Reply:
The "ExtractUtils" are global and are used across all importers.
So we would create the auxiliary class in this importer, because its here.

Reply:
you mean as inner class? I think then we can still use one additional (renamed) method...

Reply:
No, I think we're talking at cross purposes.
The function:
- rename "setDateAndAmountForAccountStatement" to "DateTransactionHelper"
- Remove the amount and currency from the function
- Replace the source code with the function

Reply:
> No, I think we're talking at cross purposes. The function:
> 
>     * Rename "setDateAndAmountForAccountStatement" to "DateTransactionHelper"

"DateTransactionHelper" sounds for me lik a static helper class. Or do you really mean a method named
"private void DateTransactionHelper(AccountTransaction t, ParsedData v)" ? Looks strange for me. 

>     * Remove the amount and currency from the function

no problem...

>     * Replace the source code with the function

is also clear to me.

Reply:
C'mom... Do we really want to make it that complicated?
- Rename "setDateAndAmountForAccountStatement" to "DateTransactionHelper"
---> `private void DateTransactionHelper(AccountTransaction t, ParsedData v)`

If you look at other importers, such as Comdirect, we handle it the same way there. The name reflects exactly what the function is supposed to do.

Well then... Let's go 👍🏻 
Alex

Reply:
Sorry but I've never seen a method named starting with capital letter and ended with *Helper... Also not in the Comdirect Importer. This will give an SonarLint warning. 
But new proposal is committed now :)